code/espurna/garland/anim.cpp

```diff
@@ -0,0 +1,38 @@
+#if GARLAND_SUPPORT

```

Comment:
Hmmm... Note that this won't build unless it's either single-source build or -DGARLAND_SUPPORT=1 in the flags.
(and why I wanted this as a lib :)

PIO also accidentally picks up Adafruit lib via code/espurna/garland/scene.cpp when resolving build deps, but it's not really working anyways and is something that I wanted to fix anyway

Reply:
I agree, that adding Adafruit_Neopixel to the whole libs set just for garland feature is redundant. And also can cause unwanted side effects as similar library already included. I can move most files from garland folder to the library, that can hosted on github.com/ElderJoy/Garland, now it way more ready to movement.
But I prefer leave anims in garland folder inside espurna project as examples and to encourage community write their own animations.
And I don't really know how to include Adafruit_Neopixel just for garland except placing it as lib_deps for specially added [env:garland-wemos].
So, please tell me what to do better.

Reply:
> I agree, that adding Adafruit_Neopixel to the whole libs set just for garland feature is redundant. And also can cause unwanted side effects as similar library already included. I can move most files from garland folder to the library, that can hosted on github.com/ElderJoy/Garland, now it way more ready to movement.

The gist of the PIO feature is https://docs.platformio.org/en/latest/librarymanager/ldf.html btw
But, I have not had much luck making it ignore any libs specified in lib_deps based on #define or -D... flags. It seems to only work correctly when:
- lib_deps is empty
- lib is somehow installed in either ~/.platformio/lib/$libname or the location of lib_extra_dirs=...
.pio/libdeps/$env/... surprisingly does not work
- project's .h #include library header (for some reason even with the deep+, main.cpp including the lib is ignored :/)

Otherwise, it simply builds everything in lib_deps list regardless of the things it finds via the scanner. I do agree it is better to keep animations in the repo though. Then, not sure it is really worth then splitting. atm I'd focus on adafruit class used multiple times. So, either:
- scene.cpp contents can be moved into garland.cpp
- the interface of the scene should not refer to the pixel class (e.g. to link with garland.cpp.o class implementing wrappers calling pixel.show(), pixel.set...(), etc.)

> But I prefer leave anims in garland folder inside espurna project as examples and to encourage community write their own animations.

The 'sensor' approach I mentioned also avoids another .cpp.o to link, by making animation classes header-only - moving anim.h declaration and AnimSomething.h into one file, then either garland.cpp or anim.h includes everything available. It also allows top-level `#if GARLAND_SUPPORT` to work correctly. Dummy .o file is still built though (~57kb in size), not sure what to do about that just yet.

Reply:
Ok, then I making next:
1. moving scene.cpp into garland.cpp
2. other sources from garland folder combining into one header garland.h and place next to garland.cpp

I totally agree with 1 but a bit doubt about 2. Combining all sources into one huge header decrease its readability, maintainability and extensibility. In case when two community members add their animations they probably would forced to merge this header. It's not a big deal but I prefer avoid that if possible. 
But if it is much better from whole project perspective and leads to smaller binary size and proper working of #if GARLAND_SUPPORT I agree to do that. Should I?

Reply:
Not all though, only animation part. What I meant about the animation headers was:
- `garland/animation.h` with basic interface
- `garland/animation/comet.h` (or `garland/AnimationComet.h`)
- `garland/animation/assemble.h
- and etc.

Reply:
Ok, got it. Looks reasonable. Will do in that way and then commit new bunch of changes. However, I'm adding one more feature now - animation speed and possibility to control it from UI. So, commit after finishing it. Soon :)